### PR TITLE
cls/rgw: Clean up the "magic string" usage in the cls layer for RGW.

### PR DESCRIPTION
**Pull request by ilc**
This set of changes, should make following what is going on across the cls
barrier much easier for people using code browsing tools.

Signed-off-by: Ira Cooper <ira@redhat.com>

### CODE REVIEW

File: src/cls/rgw/cls_rgw_const.h [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+#define RGW_GC_REMOVE "gc_remove"
+
+/* lifecycle bucket list */
+#define RGW_LC_SET_ENTRY "lc_set_entry"
+#define RGW_LC_RM_ENTRY "lc_rm_entry"
+#define RGW_LC_GET_NEXT_ENTRY "lc_get_next_entry"
+#define RGW_LC_PUT_HEAD "lc_put_head"
+#define RGW_LC_GET_HEAD "lc_get_head"
+#define RGW_LC_LIST_ENTRIES "lc_list_entries"
+
```
Comment (adamemerson) [inline src/cls/rgw/cls_rgw_const.h]: I might be tempted to favor constexpr over define myself for these, but that's a matter of taste more than anything.

**Comment by liewegas** [comment]
looks ok to me

File: src/cls/rgw/cls_rgw_const.h (hunk 1 above)
Comment (mattbenjamin) [inline src/cls/rgw/cls_rgw_const.h]: Just to clarify, a constexpr array of char, or something else?  (I'm still in the pre-processor era, for this application...)

**Comment by ilc** [comment]
In this case... I'd like to leave the header file open to C users, if the APIs to use it exist.  (Even if not, they may in the future.)  Remember we have librgwfile already, and other C libraries.

**Comment by ilc** [comment]
Can we get this merged?

**Comment by mattbenjamin** [comment]
@ilc probably mid next week, hipriority has been closing out final K blockers (one merged today)

**Comment by tchaikov** [comment]
@mattbenjamin  ping?

**Comment by tchaikov** [comment]
@mattbenjamin ping?

**Comment by cbodley** [comment]
thanks @ilc, sorry for the delay

**Pull request merged**